ecosystem/node-checker/src/system_information_evaluator/build_version_evaluator.rs

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO: Sometimes build_commit_hash is an empty string (so far I've noticed
+// this happens when targeting a node running from a container). Figure out
+// what to do in this case.
+
+use super::{
+    get_value, GetValueResult, SystemInformationEvaluator, SystemInformationEvaluatorError,
+    EVALUATOR_SOURCE,
+};
+use crate::{evaluator::EvaluationResult, metric_collector::SystemInformation};
+use anyhow::Result;
+use clap::Parser;
+use log::debug;
+use poem_openapi::Object as PoemObject;
+use serde::{Deserialize, Serialize};
+
+pub const BUILD_VERSION_EVALUATOR_NAME: &str = "build_commit_hash";
+
+// TODO: Use the key in crates/aptos-telemetry/src/build_information.rs
+const BUILD_COMMIT_HASH_KEY: &str = "build_commit_hash";
+
+#[derive(Clone, Debug, Deserialize, Parser, PoemObject, Serialize)]
+pub struct BuildVersionEvaluatorArgs {}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+pub struct BuildVersionEvaluator {
+    args: BuildVersionEvaluatorArgs,
+}
+
+impl BuildVersionEvaluator {
+    pub fn new(args: BuildVersionEvaluatorArgs) -> Self {
+        Self { args }
+    }
+
+    fn get_build_commit_hash(&self, system_information: &SystemInformation) -> GetValueResult {
+        let evaluation_on_missing_fn = || EvaluationResult {
+            headline: "State sync version metric missing".to_string(),
```

Comment:
Copy pasta ?😄

Reply:
Dang I fixed this but I guess in a later PR hahah.